packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -4472,21 +4493,51 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
           node.override ??= false;
           node.optional ??= false;
           return;
-        case "ClassDeclaration":
         case "ClassExpression":
+          node.id ??= null;
+        /* fallthrough */
+        case "ClassDeclaration":
           node.abstract ??= false;
           node.declare ??= false;
           node.decorators ??= [];
           node.implements ??= [];
           node.superTypeArguments ??= undefined;
           node.typeParameters ??= undefined;
           return;
+        case "TSTypeAliasDeclaration":
         case "VariableDeclaration":
           node.declare ??= false;
           return;
         case "VariableDeclarator":
           node.definite ??= false;
           return;
+        case "TSEnumDeclaration":
+          node.const ??= false;
+          node.declare ??= false;
+          return;
+        case "TSEnumMember":
+          node.computed ??= false;
+          return;
+        case "TSImportType":
+          node.qualifier ??= null;
+          node.options ??= null;
+          if (process.env.BABEL_8_BREAKING) {
+            node.typeArguments ??= null;
+          }
+          return;
+        case "TSInterfaceDeclaration":
+          node.declare ??= false;
+          node.extends ??= [];
+          return;
+        case "TSModuleDeclaration":
+          node.declare ??= false;
+          node.global ??= node.kind === "global";
```

Comment:
Since both we and they plan to remove it in the next major version, what do you think about not adding it?

Reply:
If we remove it before they release the next major, we will have to add those impacted tests to the failure set. However, test in the failure set does not really assert _why_ the AST is different: So it may hide more AST differences than we expected.

For the time being, let's keep it for the compatibility with the _current_ ts-eslint. We can remove it once they release a new major.